android/java/org/chromium/chrome/browser/crypto_wallet/util/KeystoreHelper.java

```diff
@@ -0,0 +1,119 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.util;
+
+import android.security.keystore.KeyGenParameterSpec;
+import android.security.keystore.KeyProperties;
+import android.util.Base64;
+
+import org.chromium.chrome.browser.preferences.BravePreferenceKeys;
+import org.chromium.chrome.browser.preferences.SharedPreferencesManager;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.CertificateException;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+
+public class KeystoreHelper {
+    private static final String TRANSFORMATION = "AES/GCM/NoPadding";
+    private static final String BRAVE_WALLET_ALIAS = "BRAVE_WALLET_ALIAS";
+    private static final String ANDROID_KEY_STORE = "AndroidKeyStore";
+
+    public static void useBiometricOnUnlock(String text) {
+        if (!encryptText(text)) {
+            return;
+        }
+
+        SharedPreferencesManager.getInstance().writeBoolean(
+                BravePreferenceKeys.BRAVE_USE_BIOMETRICS_FOR_WALLET, true);
+    }
+
+    private static boolean encryptText(String text) {
+        try {
+            Cipher cipher = Cipher.getInstance(TRANSFORMATION);
+            cipher.init(Cipher.ENCRYPT_MODE, getSecretKey());
+            byte[] iv = cipher.getIV();
+            byte[] encrypted = cipher.doFinal(text.getBytes("UTF-8"));
+            saveToSharedPref(iv, encrypted);
+        } catch (Exception exc) {
+            // Fallback to password only if we fail to encrypt it for biometric
+            return false;
+        }
+
+        return true;
+    }
+
+    private static SecretKey getSecretKey() throws NoSuchAlgorithmException, NoSuchProviderException
+                                                   ,
```

Comment:
seems this formatting is from `git cl format`

Reply:
yup, it corrected to the above. Looks ugly though